src/app_service/service/community_tokens/service.nim

```diff
@@ -286,34 +326,118 @@ QtObject:
+    if len(tokenIds) == 0:
+      error "Can't find token ids to burn"
+    return tokenIds
+
+  # TODO use temp fees for deployment also
+  proc buildTransactionFromTempFees(self: Service, addressFrom: string): TransactionDataDto =
+    return ens_utils.buildTransaction(parseAddress(addressFrom), 0.u256, $self.tempGasUnits,
+      if self.tempSuggestedFees.eip1559Enabled: "" else: $self.tempSuggestedFees.gasPrice, self.tempSuggestedFees.eip1559Enabled,
+      if self.tempSuggestedFees.eip1559Enabled: $self.tempSuggestedFees.maxPriorityFeePerGas else: "",
+      if self.tempSuggestedFees.eip1559Enabled: $self.tempSuggestedFees.maxFeePerGasM else: "")
+
+  proc selfDestructCollectibles*(self: Service, communityId: string, password: string, walletAndAmounts: seq[WalletAndAmount], contractUniqueKey: string) =
+    try:
+      let contract = self.findContractByUniqueId(contractUniqueKey)
+      let tokenIds = self.getTokensToBurn(walletAndAmounts, contract)
+      if len(tokenIds) == 0:
+        return
+      let addressFrom = self.contractOwner(contract.chainId, contract.address)
+      let txData = self.buildTransactionFromTempFees(addressFrom)
+      debug "Remote destruct collectibles ", chainId=contract.chainId, address=contract.address, tokens=tokenIds
+      let response = tokens_backend.remoteBurn(contract.chainId, contract.address, %txData, password, tokenIds)
+      let transactionHash = response.result.getStr()
+      debug "Remote destruct transaction hash ", transactionHash=transactionHash
+
+      var data = RemoteDestructArgs(communityToken: contract, transactionHash: transactionHash, status: TransactionStatus.Pending)
+      self.events.emit(SIGNAL_REMOTE_DESTRUCT_STATUS, data)
+
+      # observe transaction state
+      self.transactionService.watchTransaction(
+        transactionHash,
+        addressFrom,
+        contract.address,
+        $PendingTransactionTypeDto.CollectibleRemoteSelfDestruct,
+        $contract.toJsonNode(),
+        contract.chainId,
+      )
+    except Exception as e:
+      error "Remote self destruct error", msg = e.msg
+
+  proc computeSelfDestructFee*(self: Service, walletAndAmountList: seq[WalletAndAmount], contractUniqueKey: string) =
+    let contract = self.findContractByUniqueId(contractUniqueKey)
+    let tokenIds = self.getTokensToBurn(walletAndAmountList, contract)
+    if len(tokenIds) == 0:
+      warn "token list is empty"
+      return
+    try:
+      let arg = AsyncGetBurnFees(
+        tptr: cast[ByteAddress](asyncGetBurnFeesTask),
+        vptr: cast[ByteAddress](self.vptr),
+        slot: "onSelfDestructFees",
+        chainId: contract.chainId,
+        contractAddress: contract.address,
+        tokenIds: tokenIds
+      )
+      self.threadpool.start(arg)
+    except Exception as e:
+      error "Error loading fees", msg = e.msg
+
+  proc createComputeFeeArgs(self:Service, jsonNode: JsonNode, gasUnits: int): ComputeFeeArgs =
```

Comment:
Might be worth putting the content of this function in a `try` too, since we handle JSON

Reply:
I create the json in async_task result so I am sure it is ok.